standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -1335,18 +1335,42 @@ private void create_database_core(RawStore ms, final Database db)
   }
 
   @Override
+  @Deprecated
   public void create_database(final Database db)
       throws AlreadyExistsException, InvalidObjectException, MetaException {
-    startFunction("create_database", ": " + db.toString());
+    CreateDatabaseRequest req = new CreateDatabaseRequest(db.getName());
+    req.setDescription(db.getDescription());
+    req.setLocationUri(db.getLocationUri());
+    req.setParameters(db.getParameters());
+    req.setPrivileges(db.getPrivileges());
+    req.setOwnerName(db.getOwnerName());
+    req.setOwnerType(db.getOwnerType());
+    req.setCatalogName(db.getCatalogName());
+    req.setCreateTime(db.getCreateTime());
+    req.setManagedLocationUri(db.getManagedLocationUri());
+    req.setType(db.getType());
+    req.setDataConnectorName(db.getConnector_name());
+    req.setRemote_dbname(db.getRemote_dbname());
+
+    create_database_req(req);
+    //location and manged location might be set/changed.
+    db.setLocationUri(req.getLocationUri());
```

Comment:
Why do we change the `db` after we flush the changes, is there somewhere we must use the `db` afterwards?

Reply:
I actually thought that tests are using the `db` to assert on the location but looks like none of the tests are doing it. My only concern is, the old client, may be expecting a modified location that will no more be compatible with this change. So, if we are not removing the old/deprecated APIs in this version then it is safe to keep this or else we can remove it otherwise.

Reply:
The method is declared as void, the old remote client does not expect this method to return anything, it couldn't reach out the the modified `db` unless using a get database request.
I saw from the old `create_database_core` method:
https://github.com/apache/hive/blob/master/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java#L1151

it doesn't reset the location after flushing `db`.

Reply:
If we need to restore the location of the input `db`, we can create a new one based on the input:
`Database database = new Database(db)`